src/video.cpp

```diff
@@ -1343,6 +1352,12 @@ namespace video {
             BOOST_LOG(error) << "Could not convert image"sv;
             return;
           }
+          else {
+            if (dummy_img) {
+              // free dummy image as we don't need it anymore
+              dummy_img.reset();
```

Comment:
We actually don't even need to wait until here to free it. We can free it right after `convert()`.

I did that in #1136

Reply:
Makes sense. One thing though, I still haven't verified that image destruction during capture is implemented correctly across all platforms. And on Linux three capture backends use `snapshot()` in `dummy_img()`. So I can cherry-pick your commit into this pull request and test it together, or you merge and I rebase. Your call, either is fine with me.

Reply:
I took a look through them and it looks like it should be safe. Capture backends that support `PARALLEL_ENCODING` need to have images that are multi-threading safe (as long as they aren't actually being captured and encoded at the same time, which the parallel encoding code prevents with the `use_count()` checks).

Capture might be happening concurrently, but we know that the particular image we're about to free is not going to be involved in capture or encoding.

Reply:
Cherry-picked into this pull request.